docs/jsonapi-spec.textile

```diff
@@ -408,12 +502,33 @@ h4(#commandFindOneAndUpdateOptions). findOneAndUpdate Command Options
 | @returnDocument@ | String Enum | Specifies which document to perform the projection on. If @"before"@ the projection is performed on the document before the update is applied, if @"after"@ the document projection is from the document after the update. Defaults to @"before"@.|
 | @upsert@ | Boolean | When @true@ if no documents match the @filter@ clause the command will create a new _empty_ document and apply the @update@ clause to the empty document. If the @_id@ field is included in the @filter@ the new document will use this @_id@, otherwise a random value will be used see "Upsert Considerations":#considerationsUpsert for details. When false the command will only update a document if one matches the filter. Defaults to @false@.|
 
+h4(#commandFindOneAndUpdateResponse). findOneAndUpdate Command Response
+
+|_. Response Element |_. Description |
+| @data@ | Present with fields : @data@ only |
+| @status@ |Preset with fields: @upsertedId: <json-value>@ only if a document was upserted. |
+| @errors@ |Present if errors occur. |
+
+If @upsert@ option was set to @true@, and no documentds matched a filter a new document is created. The @_id@ of the document is included in the status field @upsertedId@, otherwise no status is returned .
+
+If an error occurs the command will not return @data@ or @status@. 
+
 h3(#commandFindOne). findOne Command
 
 h4(#commandFindOneOptions). findOne Command Options
 
 The @findOne@ command does not support any options.
 
+h4(#commandFindOneResponse). findOne Command Response
+
+|_. Response Element |_. Description |
+| @data@ | Present with fields : @docs@ only |
```

Comment:
so find one has no count? Is it ok that find commands have different responses?

Reply:
We decided to only include fields when there is useful information, so dropped count. Honestly not sure we need count anywhere, it is only the size of the `docs` array.